### PR TITLE
feat(dds): add new user resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new database user resource support.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dds' TESTARGS='-run=TestAccDatabaseUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds -v -run=TestAccDatabaseUser_basic -timeout 360m -parallel 4
=== RUN   TestAccDatabaseUser_basic
=== PAUSE TestAccDatabaseUser_basic
=== CONT  TestAccDatabaseUser_basic
--- PASS: TestAccDatabaseUser_basic (1008.92s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds 1009.021s
```

### CODE REVIEW

File: docs/resources/dds_database_user.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+* `password` - (Required, String) Specifies the user password.
+  The assword content must meet the following conditions:
+  + `8` to `32` characters long.
+  + Must contains uppercase and lowercase letters, digits, and at least one special character (`~!@#%^*-_=+?`).
+
+* `roles` - (Required, List, ForceNew) Specifies the list of roles owned by the current user.
+  The [object](#dds_database_owned_roles) structure is documented below. Changing this parameter will create a new user.
+
+* `db_name` - (Optional, String, ForceNew) Specifies the database name to which the user belongs. Defaults to **admin**.
```
Comment (hmbig2) [inline docs/resources/dds_database_user.md]: `Required` maybe better.

File: huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+	instanceId := d.Get("instance_id").(string)
+	config.MutexKV.Lock(instanceId)
+	defer config.MutexKV.Unlock(instanceId)
+
+	// Before deleting database user, we need to ensure that the database is not currently performing operations.
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"ACTIVE"},
+		Refresh:      instanceActionsRefreshFunc(client, instanceId),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
```
Comment (hmbig2) [inline huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go]: s/schema.TimeoutDelete/schema.TimeoutCreate

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dds' TESTARGS='-run=TestAccDatabase'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds -v -run=TestAccDatabase -timeout 360m -parallel 4
=== RUN   TestAccDatabaseRole_basic
=== PAUSE TestAccDatabaseRole_basic
=== RUN   TestAccDatabaseUser_basic
=== PAUSE TestAccDatabaseUser_basic
=== CONT  TestAccDatabaseRole_basic
=== CONT  TestAccDatabaseUser_basic
--- PASS: TestAccDatabaseRole_basic (838.96s)
--- PASS: TestAccDatabaseUser_basic (891.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds 891.404s
```

File: docs/resources/dds_database_user.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/dds_database_user.md]: seems great!

File: huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go]: done.

File: docs/resources/dds_database_user.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+* `roles` - (Required, List, ForceNew) Specifies the list of roles owned by the current user.
+  The [object](#dds_database_owned_roles) structure is documented below. Changing this parameter will create a new user.
+
+<a name="dds_database_owned_roles"></a>
+The `roles` block supports:
+
+* `name` - (Required, String, ForceNew) Specifies the name of role owned by the current user.
+  The name can contain `1` to `64` characters, only letters, digits, underscores (_), hyphens (-) and dots (.) are
+  allowed. Changing this parameter will create a new role.
```
Comment (Jason-Zhang9309) [inline docs/resources/dds_database_user.md]: s/new role/new user

File: docs/resources/dds_database_user.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+* `db_name` - (Required, String, ForceNew) Specifies the database name to which this owned role belongs.
+  Changing this parameter will create a new role.
```
Comment (Jason-Zhang9309) [inline docs/resources/dds_database_user.md]: ditto

File: huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+	client, err := conf.DdsV3Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating DDS v3 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	config.MutexKV.Lock(instanceId)
+	defer config.MutexKV.Unlock(instanceId)
+
+	// Before deleting database user, we need to ensure that the database is not currently performing operations.
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go]: Before creating

File: docs/resources/dds_database_user.md (hunk 4 above)
Comment (Lance52259) [reply docs/resources/dds_database_user.md]: done.

File: docs/resources/dds_database_user.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/dds_database_user.md]: done.

File: huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/dds/resource_huaweicloud_dds_database_user.go]: done.

File: docs/resources/dds_database_user.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+* `id` - The resource ID.
+
+* `privileges` - The list of database privileges owned by the current user.
+  The [object](#dds_database_privileges) structure is documented below.
+
+* `inherited_privileges` - The list of database privileges owned by the current user, includes all privileges
+  inherited by owned roles. The [object](#dds_database_privileges) structure is documented below.
+
+<a name="dds_database_privileges"></a>
+The `privileges` block supports:
```
Comment (ShiChangkuo) [inline docs/resources/dds_database_user.md]: the block is used by `privileges` and `inherited_privileges`

File: docs/resources/dds_database_user.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 2 minute.
+* `update` - Default is 2 minute.
+* `delete` - Default is 2 minute.
+
+## Import
+
+Database users can be imported using their `id` (combination of `instance_id`, `db_name` and `name`), separated by a
```
Comment (ShiChangkuo) [inline docs/resources/dds_database_user.md:108]: it seems that `password` can not be imported.

Comment (Lance52259) [reply docs/resources/dds_database_user.md:108]: done.

File: docs/resources/dds_database_user.md (hunk 6 above)
Comment (Lance52259) [reply docs/resources/dds_database_user.md]: done.

**Pull request merged**